zio-http/src/main/scala/zhttp/http/Cookie.scala

```diff
@@ -120,6 +140,30 @@ final case class Cookie(
     cookie.flatten.mkString("; ")
   }
 
+  /**
+   * Signs cookie content with a secret
+   */
+  private def signContent(secret: String): String = {
+    val sha256    = Mac.getInstance("HmacSHA256")
+    val secretKey = new SecretKeySpec(secret.getBytes(), "RSA")
+    sha256.init(secretKey)
+    val signed    = sha256.doFinal(self.content.getBytes())
+    val mda       = MessageDigest.getInstance("SHA-512")
+    self.content + '.' + getEncoder.encodeToString(mda.digest(signed))
+  }
+
+  /**
+   * Unsigns cookie content with a secret
+   */
+  private def unSignContent(secret: String): Option[Cookie] = {
+    val str             = self.content.slice(0, content.lastIndexOf('.'))
+    val unSignedCookie  = self.withContent(str)
+    val encryptedCookie = unSignedCookie.signContent(secret)
+
+    if (encryptedCookie == self.content)
+      Some(unSignedCookie.sign(secret))
+    else None
+  }
```

Comment:
private method/functions should either return null or throw an error.  In this case we should throw a error.